companion/src/mainwindow.cpp

```diff
@@ -69,12 +68,16 @@
   #define COMPANION_INSTALL_QUESTION      tr("Would you like to launch the installer?")
 #else
   #define COMPANION_STAMP                 "companion-linux.stamp"
-  #define COMPANION_INSTALLER             ""   // no automatated updates for linux
+  #define COMPANION_INSTALLER             "" // no automated updates for linux
   #define COMPANION_FILEMASK              "*.*"
   #define COMPANION_INSTALL_QUESTION      tr("Would you like to launch the installer?")
 #endif
 
-#define OPENTX_NIGHT_COMPANION_DOWNLOADS  "https://downloads-22.open-tx.org/nightlies/companion"
+const char * const OPENTX_COMPANION_DOWNLOAD_URL[] = {
+  "https://downloads.open-tx.org/2.2/release/companion",
+  "https://downloads.open-tx.org/2.2/rc/companion",
+  "https://downloads.open-tx.org/2.2/nightlies/companion"
+};
 
 MainWindow::MainWindow():
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this function to reduce its Cognitive Complexity from 26 to the 25 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=cpp%3AS3776)